### PR TITLE
add enum field support

### PR DESCRIPTION
**Pull request by agoelzer**
currently users are presented with a text field instead of a combo box and they need to enter exactly one of specific values. We need to present a combo box with the values instead (i.e. Backup Policies / Database Retention / Day of Week)

### CODE REVIEW

File: ui/src/components/fields/FieldSelect.tsx [hunk 1]
```diff
@@ -11,16 +11,16 @@ export default function FieldSelect(props: FieldProps): FieldBaseType {
      * show Field of type Boolean using the values and schema definition
      */
     function show(): ReactNode {
-        const { prefix, label, values, parameter, required, setValues, autoFocus, readonly } = props;
+        const { prefix, label, values, parameter, required, setValues, autoFocus, readonly, t } = props;
```
Comment (adriansuarez) [inline ui/src/components/fields/FieldSelect.tsx:14]: What is `t`?

Comment (agoelzer) [reply ui/src/components/fields/FieldSelect.tsx:14]: it is a function provided by i18next (the translation framework) to translate a key to a label. `t` is intentionally one letter so the code is not clogged with a long function name.

Comment (adriansuarez) [reply ui/src/components/fields/FieldSelect.tsx:14]: Okay, thanks for the explanation.

File: ui/src/resources/translation/de.json [hunk 2]
```diff
@@ -66,11 +66,43 @@
             "frequency": "Frequenz",
             "selector": "Auswahl"
         },
+        "enum": {
+            "theme.material": "Material UI",
+            "theme.plain": "einfach",
+            "retention.settings": {
```
Comment (adriansuarez) [inline ui/src/resources/translation/de.json:72]: Should we also have `status.state` for project, database, and backup?

**Review by adriansuarez — APPROVED** [review]
This a good improvement.

File: ui/src/resources/translation/de.json (hunk 2 above)
Comment (agoelzer) [reply ui/src/resources/translation/de.json:72]: good point. Even though `status.state` is a read only value and therefore doesn't need combo box support, it does need translation (currently the `enum key` is shown in the view and not the `label`). I'll make that change.

**Pull request merged**